src/Relude/Nub.hs

```diff
@@ -90,3 +97,27 @@ sortNub = Set.toList . Set.fromList
 -}
 unstableNub :: (Eq a, Hashable a) => [a] -> [a]
 unstableNub = HashSet.toList . HashSet.fromList
+
+{- | A generalized nub for any 'Foldable'. Not necessarily order preserving. -}
+foldNub :: (Foldable t, Ord a) => t a -> Set.Set a
+foldNub = foldr Set.insert Set.empty
+
+{- | A strict variant of 'foldNub' using 'foldl'' instead of 'foldr'-}
+foldNub' :: (Foldable t, Ord a) => t a -> Set.Set a
+foldNub' = foldl' (flip Set.insert) Set.empty
+
+{- | A generalized nub for any 'Foldable' containing 'Hashable' -}
+hashFoldNub :: (Foldable t, Eq a, Hashable a) => t a -> HashSet.HashSet a
+hashFoldNub = foldr HashSet.insert HashSet.empty
+
+{- | A strict variant of 'hashFoldNub' using 'foldl'' instead of 'foldr'-}
+hashFoldNub' :: (Foldable t, Eq a, Hashable a) => t a -> HashSet.HashSet a
+hashFoldNub' = foldl' (flip HashSet.insert) HashSet.empty
```

Comment:
Btw, instead of `foldl' (flip Set.insert)` you can use `flipfoldl'` function:

```
foldNub = flipfoldl' Set.insert Set.empty
```

And you can even use `mempty` instead of `Set.empty` for brevity if you like 🙂

Reply:
Yes, I can do this. One question I have though is about style. I'm wondering whether

```haskell
foldl' (\xs x-> Set.insert x xs) Set.empty
```
is even clearer. It's a little longer, but has little more information, so I don't know. The right balance of brevity and clarity is so tough to get right sometimes.

Reply:
@johnchandlerburnham I'm okay with writing `foldl' (\xs x-> Set.insert x xs) Set.empty` instead of using `flip`. This is library function, should be written only once, it's a not a subject of possible refactoring. Later can be only read by other people. And the definition of the function is still short even with explicit lambda function, so I'm okay with any implementation :+1: